runtime/gc_vlhgc/CopyForwardScheme.cpp

```diff
@@ -494,15 +514,14 @@ MM_CopyForwardScheme::postProcessRegions(MM_EnvironmentVLHGC *env)
 	UDATA survivorSetRegionCount = 0;
 
 	while(NULL != (region = regionIterator.nextRegion())) {
-		MM_MemoryPoolBumpPointer *pool = (MM_MemoryPoolBumpPointer *)region->getMemoryPool();
-
+		MM_MemoryPoolAddressOrderedList *pool = (MM_MemoryPoolAddressOrderedList*)region->getMemoryPool();
 		if (region->_copyForwardData._evacuateSet) {
 			if (region->isEden()) {
 				static_cast<MM_CycleStateVLHGC*>(env->_cycleState)->_vlhgcIncrementStats._copyForwardStats._edenEvacuateRegionCount += 1;
 			} else {
 				static_cast<MM_CycleStateVLHGC*>(env->_cycleState)->_vlhgcIncrementStats._copyForwardStats._nonEdenEvacuateRegionCount += 1;
 			}
-		} else if (region->isSurvivorRegion() && !region->isTailFilledSurvivorRegion()) {
+		} else if (region->isFreshSurvivorRegion()) {
```

Comment:
I don't know why are these stats important, but perhaps we could count amount of bytes (count cards, as we dirty them in survivor CT) that have been used in non-fresh regions. Ignore for now (till I better understand, if that would be useful anywhere).

Reply:
the flag for FreshSurvivorRegion is just for identifying the survior Region is from Free Region in order to simplify further check condition, if there is another easy way to verify  non-fresh regions, then we can eliminate it.

Reply:
That flag is ok. I was just commenting that we were gathering some stats for fresh regions, but not for fragmented/already occupied ones. As I said, ignore this comment now, since I don't see that stats would be much useful.